compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -903,10 +903,33 @@ std::tuple<int64_t, int64_t, int64_t> DataTiledMMAAttr::getMNKShape() const {
           opaqueLayout.kSize * getUnrollK()};
 }
 
+// Returns the swizzled tile shape, but with dim sizes overwritten with 1 if
+// `predicate` returns false.
+static SmallVector<int64_t>
+sliceSwizzledShape(const TileSwizzle &swizzle,
+                   std::function<bool(TileSwizzle::Dim)> predicate) {
+  SmallVector<int64_t> shape;
+  for (auto e : swizzle.expandShape) {
+    for (auto d : e) {
```

Comment:
nit: Use actual types here -- it's hard to know what these are based on the names only